RUFAS/biophysical/animal/ration/ration_driver.py

```diff
@@ -143,8 +143,8 @@ def formulate_ration(
             return ration, ration_vals
         # safeguard if scipy SLSQP bounds error still occurs after many iterations
         # using previous cycles ration for this pen
-        elif pen.ration != {}:
-            return pen.ration, ration_vals
+        elif pen.ration_per_animal != {}:
+            return pen.ration_per_animal, ration_vals
```

Comment:
```suggestion
        elif previous_ration:
            return previous_ration, ration_vals
```
Might be better to reuse the variable that has ration per animal. Or, if `pen.ration_per_animal` is kept it might be good to check that the previous ration is present in pen (and not None), this is accomplished earlier in the code with `hasattr`.